src/resolve/resolved-dns-packet.c

```diff
@@ -47,13 +50,14 @@ int dns_packet_new(DnsPacket **ret, DnsProtocol protocol, size_t mtu) {
 
         assert(ret);
 
-        if (mtu <= UDP_PACKET_HEADER_SIZE)
+        /* When dns_packet_new() is called with mtu == 0, allocate more than the
+         * absolute minimum (which is the dns packet header size), to avoid
+         * resizing immediately again after appending the first data to the packet.
+         */
+        if (mtu < UDP_PACKET_HEADER_SIZE)
```

Comment:
@poettering I see, it make sense, but the interface is never used like that. 
 It always set to 0 or the size of the data. That why the bug was introduced in a016660 .
The mtu include the UDP header, but here we are allocating just the size for the data, not the size of the raw packet